### PR TITLE
Add object detection enhancements and utility functions

### LINKED ISSUE #46: Remove classes from blur_image and instead just blur every box in the results (user has to filter results)

### LINKED ISSUE #47: Chained detection return tuple instead of merged results

### PR DESCRIPTION
**Pull request by TheDying0fLight**
Resolves #47
Resolves #46 

- Introduced new methods for image processing and detection chaining in the ObjectDetection class.
- Added a utility function to merge results from multiple detections.
- Updated test notebook to reflect changes in the ObjectDetection class and improve clarity.
- Configured VSCode settings for autopep8 formatting.

### CODE REVIEW

File: safe_video/number_plate_recognition/plate_recognition.py [hunk 1]
```diff
@@ -2,33 +2,50 @@
...
+
+    def crop_image(self, image: ImageInput, bbox: np.ndarray) -> np.ndarray:
+        assert len(bbox) == 4, "Array must have exactly 4 entries"
+        x1, y1, x2, y2 = bbox.astype("int")
+        return image[y1:y2, x1:x2]
+
     def map_classes_to_models(self, classes: list[str]) -> dict[int, list[int]]:
+        classes = classes.copy()
+        cls_len = len(classes)
+        if cls_len == 0: raise ValueError("No classes provided")
```
Comment (EmWhyy) [inline safe_video/number_plate_recognition/plate_recognition.py:48]:
```suggestion
        classes_len = len(classes)
        if classes_len == 0: raise ValueError("No classes provided")
```

File: safe_video/number_plate_recognition/utils.py [hunk 2]
```diff
@@ -37,6 +39,20 @@ def merge_results(result1: Results, result2: Results) -> Results:
     merged_result.boxes.data = merged_data
     return merged_result
 
+def merge_results_list(results: list[Results]) -> Results:
```
Comment (EmWhyy) [inline safe_video/number_plate_recognition/utils.py:42]:
```suggestion
def merge_multiple_detection_results(results: list[Results]) -> Results:
```

File: safe_video/number_plate_recognition/utils.py [hunk 3]
```diff
@@ -37,6 +39,20 @@ def merge_results(result1: Results, result2: Results) -> Results:
...
+
+    Returns:
+        Results: Merged YOLO result containing all bounding boxes from all results and also updated class mapping
+    """
+    merged_result: Results = None
+    for result in results: merged_result = merge_results(merged_result, result)
+    return merged_result
+
 def find_key_by_value(dictionary: dict, value: str) -> int:
     return list(dictionary.keys())[list(dictionary.values()).index(value)]
```
Comment (EmWhyy) [inline safe_video/number_plate_recognition/utils.py:58]: I could not mark it in git but change `filter_results` to `filter_detection_results_by_class` so it directly clear what is happening

**Review by EmWhyy — CHANGES_REQUESTED** [review]
Notebook is running, tested some things and it worked fine. 
After renaming I can approve it

File: safe_video/number_plate_recognition/utils.py (hunk 3 above)
Comment (TheDying0fLight) [reply safe_video/number_plate_recognition/utils.py:58]: I think the filter function should be extendend with additional parameters making the general name fitting https://github.com/Cari1111/NumberPlateRecognition/issues/49

File: safe_video/number_plate_recognition/utils.py (hunk 2 above)
Comment (TheDying0fLight) [reply safe_video/number_plate_recognition/utils.py:42]: The other function `merge_results` also merges multiple detection results which makes this name not very helpful

**Review by EmWhyy — APPROVED** [review]
works

**Pull request merged**